### DISCUSSION TITLE
Ship of Harkinian (OoT port)

### DISCUSSION THREAD

**Discussion by veerorith** [eduardofilo/RG350_adam_image#266] [Ports]
Was wondering if it would even be possible to compile the Ship of Harkinian OoT port to my 350m.  I'm very new to linux and from the research ive done if I can hook the device up to the internet and a keyboard I should be able to use the whole 'git' command thing to download and compile it on the system, but I don't have the knowledge to really tinker if things don't just work out of the box.  Also I don't have a usb c hub for it to hook up internet and a keyboard just yet so I can't really attempt it just yet, but I wanted to know if anybody has done it already, or has any resources they could throw my way about compiling programs on opendingux.  Can you run the adam image on a virtual machine to compile stuff then move it to the handheld after the fact, so I'm not playing with dongles and drivers to make the 350m connect to the internet in the first place?  

Anyways.  I'm not asking for anybody to do it for me, I'm just very out of my depth and any sort of direction would be neat.  Like if there's a discord for all this stuff, or a wiki that explains the process of compiling things specifically for opendingux/adam. 

And if anybody wants to warn me in advance that what I'm doing is impossible or silly or just not in the scope of the system thats fine too, but I'm googling my heart out and I've managed to find a basic understanding of compiling software packages but that's about it.  Anything specific to adam or opendingux just drops me in defunct forums that wont let new users register and I'm super lost. 

Thanks for reading this, I'll keep looking around for info on my own as well. Hopefully I can figure out how to make it work, and if I do I'll share what I learn so people can do it themselves, its just very obscure and my research skills are failing me pretty hard. 

EDIT: I've learned I can't really make a virtual machine for opendingux because the processor architecture is different than your normal x86 stuff?  I think?  Anyways, after breaking hyper-v for a while, I'm just trying to get it compiled directly on the 350m.  

I'm just struggling to know what files or things I need to throw onto the device before I start trying to use console commands to hopefully compile the thing.  This'll probably be much easier when I get a compatible usb to usb c and wifi dongle for it, so I can just use git commands, but yeah.

**Comment by a-dekker** [comment]
I think this page explains it well in general: http://blakesmith.me/2020/08/09/writing-software-for-the-rg350m-gaming-handheld.html

Forget about compiling on your handheld. It's too slow and has not enough memory for compiling larger projects.
I have not seen any Zelda OOT handheld port working yet, so I guess some work has still to be done. Certainly no stuff suited for beginners.

**Reply by veerorith** [reply]
Oh cool!  I'll dig through that article.  Well even if its something I'm gonna hit a wall on I'm gonna mess with it anyways, I really need to learn linux and worst case its a learning experience, I'll letcha know if I brick the handheld though lol
